src/codewhisperer/util/editorContext.ts

```diff
@@ -51,6 +51,21 @@ export function getFileName(editor: vscode.TextEditor): string {
     return fileName.substring(0, CodeWhispererConstants.filenameCharsLimit)
 }
 
+export function getFileNameForRequest(editor: vscode.TextEditor): string {
+    const fileName = path.basename(editor.document.fileName)
+
+    // For notebook files, we want to use the programming language for each cell for the code suggestions, so change
+    // the filename sent in the request to reflect that language
+    if (fileName.endsWith('.ipynb')) {
```

Comment:
What is the languageId in a ipynb file?

Reply:
It's related to the kernel, if I specify the rust kernel it's rust, if it's typescript it will correctly give me typescript.